src/schematics/cypress/index.ts

```diff
@@ -96,8 +96,9 @@ function removeFiles(options: any): Rule {
   return (tree: Tree, context: SchematicContext) => {
     context.logger.debug('Removing e2e directory');
 
-    if (tree.getDir('./e2e')) {
-      tree.delete('./e2e');
+    const rootE2eFolder = './e2e';
+    if (!tree.exists(rootE2eFolder) && tree.getDir(rootE2eFolder).subfiles.length > 0) {
```

Comment:
```suggestion
    if (!tree.exists(rootE2eFolder) && !!tree.getDir(rootE2eFolder)) {
```
Nice work! The only thing I would suggest here would be to just check for the existence of the './e2e' directory instead of the length of subfiles. I know it would be a very strange case, but if './e2e' is there, even if it is empty, we still want to remove it.

Reply:
The getDir API returns always an object. ![image](https://user-images.githubusercontent.com/1221055/95209637-e1627580-07ea-11eb-8854-56f8cdeedeea.png)

I think there is no tree API to check if directory exists.
If you want I can use the try-catch to catch the error if tree.delete throws an error (no e2e directory exists).

Reply:
Ok I pushed the change with try-catch.